presto-prometheus/src/main/java/io/prestosql/plugin/prometheus/PrometheusConnectorConfig.java

```diff
@@ -0,0 +1,128 @@
+import javax.validation.constraints.NotNull;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+public class PrometheusConnectorConfig
+{
+    private URI prometheusURI = new URI("http://localhost:9090");
+    private Duration queryChunkSizeDuration = new Duration(1, TimeUnit.DAYS);
+    private Duration maxQueryRangeDuration = new Duration(21, TimeUnit.DAYS);
+    private Duration cacheDuration = new Duration(30, TimeUnit.SECONDS);
+    private String bearerTokenFile;
+
+    public PrometheusConnectorConfig()
+            throws URISyntaxException
+    {}
+
+    @NotNull
+    public URI getPrometheusURI()
+    {
+        return prometheusURI;
+    }
+
+    @Config("prometheus-uri")
+    @ConfigDescription("Where to find Prometheus coordinator host")
+    public PrometheusConnectorConfig setPrometheusURI(URI prometheusURI)
+    {
+        this.prometheusURI = prometheusURI;
+        return this;
+    }
+
+    @MinDuration("1ms")
+    public Duration getQueryChunkSizeDuration()
+    {
+        return queryChunkSizeDuration;
+    }
+
+    public String getQueryChunkSizeDurationAsPrometheusCompatibleDurationString()
+    {
+        return String.valueOf(queryChunkSizeDuration.roundTo(queryChunkSizeDuration.getUnit())) +
+                io.airlift.units.Duration.timeUnitToString(queryChunkSizeDuration.getUnit());
+    }
+
+    @Config("query-chunk-size-duration")
+    @ConfigDescription("The duration of each query to Prometheus")
+    public PrometheusConnectorConfig setQueryChunkSizeDuration(String queryChunkSizeDuration)
+    {
+        this.queryChunkSizeDuration = Duration.valueOf(queryChunkSizeDuration);
+        return this;
+    }
+
+    @MinDuration("1ms")
+    public Duration getMaxQueryRangeDuration()
+    {
+        return maxQueryRangeDuration;
+    }
+
+    @Config("max-query-range-duration")
+    @ConfigDescription("Width of overall query to Prometheus, will be divided into query-chunk-size-duration queries")
+    public PrometheusConnectorConfig setMaxQueryRangeDuration(String maxQueryRangeDuration)
+    {
+        this.maxQueryRangeDuration = Duration.valueOf(maxQueryRangeDuration);
+        return this;
+    }
+
+    @MinDuration("1s")
+    public Duration getCacheDuration()
+    {
+        return cacheDuration;
+    }
+
+    @Config("cache.ttl")
+    @ConfigDescription("How long values from this config file are cached")
+    public PrometheusConnectorConfig setCacheDuration(String cacheConfigDuration)
+    {
+        this.cacheDuration = Duration.valueOf(cacheConfigDuration);
+        return this;
+    }
+
+    public Optional<String> getBearerTokenFile()
+    {
+        return Optional.ofNullable(bearerTokenFile);
+    }
+
+    @Config("bearer-token-file")
+    @ConfigDescription("File holding bearer token if needed for access to Prometheus")
+    public PrometheusConnectorConfig setBearerTokenFile(String bearerTokenFile)
```

Comment:
We can also use `File` type. 
```suggestion
    public PrometheusConnectorConfig setBearerTokenFile(File bearerTokenFile)
```

Reply:
Done in dccd3967